docs/sources/flow/reference/components/discovery.azure.md

```diff
@@ -0,0 +1,115 @@
+---
+title: discovery.azure
+---
+
+# discovery.azure
+
+`discovery.azure` discovers [Azure][] Virtual Machines and exposes them as targets.
+
+[Azure]: https://azure.microsoft.com/en-us
+
+## Usage
+
+```river
+discovery.azure "LABEL" {
+}
+```
+
+## Arguments
+
+The following arguments are supported:
+
+Name                | Type       | Description                                                            | Default              | Required
+------------------- | ---------- | ---------------------------------------------------------------------- | -------------------- | --------
+`environment`       | `string`   | Azure environment.                                                     | `"AzurePublicCloud"` | no
+`port`              | `number`   | Port to be appended to the `__address__` label for each target.        | `80`                 | no
+`subscription_id`   | `string`   | Azure subscription ID.                                                 |                      | no
+`refresh_interval`  | `duration` | Interval at which to refresh the list of targets.                      | `1m`                 | no
+`proxy_url`         | `string`   | HTTP proxy to proxy requests through.                                  |                      | no
+`follow_redirects`  | `bool`     | Whether redirects returned by the server should be followed.           | `true`               | no
```

Comment:
follow_redirects and enable_http2 both seem like they will actually default to false.

Reply:
This one is a bit confusing, but internally it uses the [component.common.config.DefaultHTTPClientConfig](https://github.com/grafana/agent/blob/main/component/common/config/types.go#L135), which has both of those defaulted to true.